freqtrade/optimize/hyperopt_interface.py

```diff
@@ -76,30 +81,102 @@ def generate_roi_table(params: Dict) -> Dict[int, float]:
+            'roi_t1_max': int(120 * roi_t_scale * roi_t_alpha),
+            'roi_t2_min': int(10 * roi_t_scale * roi_t_alpha),
+            'roi_t2_max': int(60 * roi_t_scale * roi_t_alpha),
+            'roi_t3_min': int(10 * roi_t_scale * roi_t_alpha),
+            'roi_t3_max': int(40 * roi_t_scale * roi_t_alpha),
+            'roi_p1_min': 0.01 * roi_p_scale * roi_p_alpha,
+            'roi_p1_max': 0.04 * roi_p_scale * roi_p_alpha,
+            'roi_p2_min': 0.01 * roi_p_scale * roi_p_alpha,
+            'roi_p2_max': 0.07 * roi_p_scale * roi_p_alpha,
+            'roi_p3_min': 0.01 * roi_p_scale * roi_p_alpha,
+            'roi_p3_max': 0.20 * roi_p_scale * roi_p_alpha,
+        }
+        logger.debug(f"Using roi space limits: {roi_limits}")
+        p = {
+            'roi_t1': roi_limits['roi_t1_min'],
+            'roi_t2': roi_limits['roi_t2_min'],
+            'roi_t3': roi_limits['roi_t3_min'],
+            'roi_p1': roi_limits['roi_p1_min'],
+            'roi_p2': roi_limits['roi_p2_min'],
+            'roi_p3': roi_limits['roi_p3_min'],
+        }
+        logger.info(f"Min roi table: {IHyperOpt.generate_roi_table(p)}")
+        p = {
+            'roi_t1': roi_limits['roi_t1_max'],
+            'roi_t2': roi_limits['roi_t2_max'],
+            'roi_t3': roi_limits['roi_t3_max'],
+            'roi_p1': roi_limits['roi_p1_max'],
+            'roi_p2': roi_limits['roi_p2_max'],
+            'roi_p3': roi_limits['roi_p3_max'],
+        }
+        logger.info(f"Max roi table: {IHyperOpt.generate_roi_table(p)}")
 
-        Defines range of stoploss values to search.
-        You may override it in your custom Hyperopt class.
-        """
         return [
-            Real(-0.5, -0.02, name='stoploss'),
+            Integer(roi_limits['roi_t1_min'], roi_limits['roi_t1_max'], name='roi_t1'),
+            Integer(roi_limits['roi_t2_min'], roi_limits['roi_t2_max'], name='roi_t2'),
+            Integer(roi_limits['roi_t3_min'], roi_limits['roi_t3_max'], name='roi_t3'),
+            Real(roi_limits['roi_p1_min'], roi_limits['roi_p1_max'], name='roi_p1'),
+            Real(roi_limits['roi_p2_min'], roi_limits['roi_p2_max'], name='roi_p2'),
+            Real(roi_limits['roi_p3_min'], roi_limits['roi_p3_max'], name='roi_p3'),
         ]
 
     @staticmethod
-    def roi_space() -> List[Dimension]:
+    def stoploss_space() -> List[Dimension]:
         """
-        Create a ROI space.
+        Create a stoploss space.
 
-        Defines values to search for each ROI steps.
+        Defines range of stoploss values to search.
         You may override it in your custom Hyperopt class.
         """
         return [
-            Integer(10, 120, name='roi_t1'),
-            Integer(10, 60, name='roi_t2'),
-            Integer(10, 40, name='roi_t3'),
-            Real(0.01, 0.04, name='roi_p1'),
-            Real(0.01, 0.07, name='roi_p2'),
-            Real(0.01, 0.20, name='roi_p3'),
+            Real(-0.5, -0.02, name='stoploss'),
```

Comment:
Maybe we should change the defaults to be less risky?

50% stoploss is quite drastic - i understand that's unlikely to be the optimum, however it would reduce the search-space if we'd reduce this to -0.4 or -0.3 (which is still quite high for my taste).

Opposed to that, i would also lower the other boundary to -0.01 - tight stoplosses CAN work for certain types of strategies (and hyperopt shall figure that out, no?)

Reply:
I'm not against this. However, I saw many hyperopt best results with stoploss=0.39, for example.
Yes, this is risky. I just kept it as it is. Let's select a new value in a sep. PR, this one is not about stoploss (I only changed the order of the methods in the class)

Reply:
absolutely agree - was just something i noticed while reading through .. (because github flags this as change, which it really isn't...)